src/services/accessValidation.test.js

```diff
@@ -1,85 +1,182 @@
-    const existingUser = await User.findOne({
-      where: {
-        hsesUserId: user.hsesUserId,
-      },
+      expect(retrievedUser.hsesUserId).toEqual(user.hsesUserId);
+      expect(retrievedUser.email).toEqual(user.email);
+      expect(retrievedUser.id).toEqual(user.id);
     });
 
-    expect(existingUser).toBeNull();
+    it('Creates a new user when a matching user does not exist', async () => {
+      const user = {
+        id: 34,
+        hsesUserId: '34',
+        email: 'test34@test.com',
+        homeRegionId: 3,
+      };
+      // Check that the above `user` doesn't exist in the DB yet.
+      await User.destroy({ where: { id: 34 } });
+      const existingUser = await User.findOne({
+        where: {
+          hsesUserId: user.hsesUserId,
+        },
+      });
 
-    // Now find or create `user2`, and confirm that a new user was created
-    const createdUser = await findOrCreateUser(user);
+      expect(existingUser).toBeNull();
 
-    expect(createdUser.id).toBeDefined();
-    expect(createdUser.email).toEqual(user.email);
+      // Now find or create `user2`, and confirm that a new user was created
+      const createdUser = await findOrCreateUser(user);
 
-    // Look up the user that was just created, make sure it can now be found
-    const existingUserAfter = await User.findOne({
-      where: {
-        id: createdUser.id,
-      },
+      expect(createdUser.id).toBeDefined();
+      expect(createdUser.email).toEqual(user.email);
+
+      // Look up the user that was just created, make sure it can now be found
+      const existingUserAfter = await User.findOne({
+        where: {
+          id: createdUser.id,
+        },
+      });
+      expect(existingUserAfter).toBeInstanceOf(User);
+    });
+
+    it('Throws when there is something wrong', async () => {
+      await expect(() => findOrCreateUser(undefined)).rejects.toBeInstanceOf(Error);
     });
-    expect(existingUserAfter).toBeInstanceOf(User);
-  });
 
-  it('Throws when there is something wrong', async () => {
-    await expect(() => findOrCreateUser(undefined)).rejects.toBeInstanceOf(Error);
+    it('Logs an error message on error', async () => {
+      const user = {
+        hsesUserId: '33',
+        email: 'invalid',
+        homeRegionId: 3,
+      };
+      await expect(findOrCreateUser(user)).rejects.toThrow();
+      expect(logger.error).toHaveBeenCalledWith('SERVICE:ACCESS_VALIDATION - Error finding or creating user in database - SequelizeValidationError: Validation error: Validation isEmail on email failed');
+    });
   });
+  describe('validateUserAuthForAdmin', () => {
+    it('returns true if a user has admin priviledges', async () => {
+      await User.destroy({ where: { id: mockUser.id } });
+      await sequelize.transaction((transaction) => User.create(mockUser,
+        {
+          include: [{ model: Permission, as: 'permissions' }],
+          transaction,
+        }));
+
+      expect(ADMIN).toBe(2);
```

Comment:
I think this will always be true

Reply:
True. I can remove if it's easier to read.